plugin.rb

```diff
@@ -567,14 +570,74 @@ def self.enqueue_assign_hooks(event, payload)
       script do |context, fields|
         next unless SiteSetting.assign_enabled?
 
+        next unless topic_id = fields.dig('assigned_topic', 'text')
+        next unless topic = Topic.find_by(id: topic_id)
+
         next unless group_id = fields.dig('assignees_group', 'group_id')
         next unless group = Group.find_by(id: group_id)
-        assign_to = group.group_users.order(Arel.sql('RANDOM()')).first.user
 
-        next unless topic_id = fields.dig('assigned_topic', 'text')
-        next unless topic = Topic.find_by(id: topic_id)
+        min_hours = (fields.dig('minimum_time_between_assignments', 'text') || 12).to_i
+        if TopicCustomField
+            .where(name: 'assigned_to_id', topic_id: topic_id)
+            .where('created_at < ?', min_hours.hours.ago)
+            .exists?
+          next
+        end
+
+        users_on_holiday = User
```

Comment:
I recommend using a `Set.new` for this. It'll make the `include?` below significantly faster if there are many users.